### PR TITLE
deviantart-integration

### PR DESCRIPTION
**Pull request by arniesaha**
Deviantart-source-integration — first commit

Hi Nick,

Here's the pull-request for issue with a new source integration described here https://github.com/nickbutcher/plaid/issues/177

Would really appreciate it if you had a look. Had a issue with CLA with last commit. Hence, re-submitting this. 

Regards,
Arnab

### CODE REVIEW

File: gradle.properties [hunk 1]
```diff
@@ -11,14 +11,19 @@
 org.gradle.jvmargs=-Xmx1536M
 org.gradle.daemon=false
 
+
 # Dribbble API
-dribbble_client_id = <your dribbble client id>
-dribbble_client_secret = <your dribbble client secret>
-dribbble_client_access_token = <your dribbble client acces token>
+dribbble_client_id = 32b4a0c2df0c7ab002bde5d1fb3c1530b8b15fadba8880431b2f200124c44664
+dribbble_client_secret = 211d111fed4e438f72c1d561d44f4ef11f07220edfe3b1abc70339c00a6248e9
```
Comment (piotrek1543) [inline gradle.properties]: Please remove all your personal keys. Restore default values.

File: app/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -106,14 +121,30 @@
             <intent-filter>
                 <data
                     android:host="dribbble-auth-callback"
-                    android:scheme="plaid" />
+                    android:scheme="foshizzle" />
```
Comment (piotrek1543) [inline app/src/main/AndroidManifest.xml]: What is 'foshizzle'? Is your account name? Is it possible to restore it to default value?

File: app/src/main/java/io/plaidapp/data/BaseDataManager.java [hunk 3]
```diff
@@ -29,11 +14,13 @@
 import io.plaidapp.data.api.AuthInterceptor;
 import io.plaidapp.data.api.DenvelopingConverter;
 import io.plaidapp.data.api.designernews.DesignerNewsService;
+import io.plaidapp.data.api.deviantart.DeviantartService;
 import io.plaidapp.data.api.dribbble.DribbbleSearchConverter;
 import io.plaidapp.data.api.dribbble.DribbbleSearchService;
 import io.plaidapp.data.api.dribbble.DribbbleService;
 import io.plaidapp.data.api.producthunt.ProductHuntService;
 import io.plaidapp.data.prefs.DesignerNewsPrefs;
+import io.plaidapp.data.prefs.DeviantartPrefs;
```
Comment (piotrek1543) [inline app/src/main/java/io/plaidapp/data/BaseDataManager.java]: Restore deleted content - comment above

File: app/src/main/java/io/plaidapp/ui/DribbbleShot.java [hunk 4]
```diff
@@ -434,6 +435,7 @@ public void run() {
         @Override
         public void onClick(View view) {
             openLink(shot.url);
+            Log.d("hiRes", shot.images.toString());
```
Comment (piotrek1543) [inline app/src/main/java/io/plaidapp/ui/DribbbleShot.java]: Create a `TAG` like above using `logt` Android Studio Live Templates and use it in this `Log.d` method.

File: app/src/main/java/io/plaidapp/ui/HomeActivity.java [hunk 5]
```diff
@@ -142,22 +146,38 @@ protected void onCreate(Bundle savedInstanceState) {
         setExitSharedElementCallback(FeedAdapter.createSharedElementReenterCallback(this));
 
         dribbblePrefs = DribbblePrefs.get(this);
-        designerNewsPrefs = DesignerNewsPrefs.get(this);
+        deviantartPrefs = DeviantartPrefs.get(this);
+//        designerNewsPrefs = DesignerNewsPrefs.get(this);
```
Comment (piotrek1543) [inline app/src/main/java/io/plaidapp/ui/HomeActivity.java]: Please remove all commented out content from this class

File: app/src/main/res/values/strings.xml [hunk 6]
```diff
@@ -18,11 +18,12 @@
 <resources>
 
     <!-- global -->
-    <string name="app_name">Plaid</string>
+    <string name="app_name">Foshizzle</string>
```
Comment (piotrek1543) [inline app/src/main/res/values/strings.xml]: Heh, please restore default name if you want @nickbutcher to merge it ;-)

File: app/src/main/res/menu/main.xml [hunk 7]
```diff
@@ -35,10 +36,17 @@
...
         android:showAsAction="never" />
 
+
+
     <item
+        android:id="@+id/menu_deviantart_login"
+        android:title="@string/deviantart_login"
+        android:showAsAction="never" />
+
+    <!--item
```
Comment (piotrek1543) [inline app/src/main/res/menu/main.xml]: Remove this comment

File: app/src/main/java/io/plaidapp/ui/HomeActivity.java [hunk 8]
```diff
@@ -490,23 +537,23 @@ public void onScrollStateChanged(RecyclerView recyclerView, int newState) {
...
-        }
+//        if (designerNewsPrefs.isLoggedIn()) {
+//            Intent intent = new Intent(this, PostNewDesignerNewsStory.class);
+//            FabTransform.addExtras(intent,
+//                    ContextCompat.getColor(this, R.color.accent), R.drawable.ic_add_dark);
+//            intent.putExtra(PostStoryService.EXTRA_BROADCAST_RESULT, true);
+//            registerPostStoryResultListener();
+//            ActivityOptions options = ActivityOptions.makeSceneTransitionAnimation(this, fab,
+//                    getString(R.string.transition_new_designer_news_post));
+//            startActivityForResult(intent, RC_NEW_DESIGNER_NEWS_STORY, options.toBundle());
```
Comment (piotrek1543) [inline app/src/main/java/io/plaidapp/ui/HomeActivity.java]: Please remove this comment

File: app/src/main/java/io/plaidapp/ui/HomeActivity.java [hunk 9]
```diff
@@ -347,17 +375,24 @@ public boolean onOptionsItemSelected(MenuItem item) {
...
+                if(!deviantartPrefs.isLoggedIn()){
+                    deviantartPrefs.login(HomeActivity.this);
+                }else{
+                    deviantartPrefs.logout();
+                    Toast.makeText(getApplicationContext(), "Logged out of Deviantart", Toast
+                          .LENGTH_LONG).show();
                 }
                 return true;
+//            case R.id.menu_designer_news_login:
+//                if (!designerNewsPrefs.isLoggedIn()) {
```
Comment (piotrek1543) [inline app/src/main/java/io/plaidapp/ui/HomeActivity.java]: Please remove this comment

File: app/src/main/java/io/plaidapp/ui/HomeActivity.java [hunk 10]
```diff
@@ -318,11 +340,17 @@ public boolean onPrepareOptionsMenu(Menu menu) {
...
-        if (designerNewsLogin != null) {
-            designerNewsLogin.setTitle(designerNewsPrefs.isLoggedIn() ?
-                    R.string.designer_news_log_out : R.string.designer_news_login);
+        final MenuItem deviantartLogin = menu.findItem(R.id.menu_deviantart_login);
+        if (deviantartLogin != null) {
+            deviantartLogin.setTitle(deviantartPrefs.isLoggedIn() ?
+                    R.string.deviantart_log_out : R.string.deviantart_login);
         }
+
+//        final MenuItem designerNewsLogin = menu.findItem(R.id.menu_designer_news_login);
```
Comment (piotrek1543) [inline app/src/main/java/io/plaidapp/ui/HomeActivity.java]: Please remove this comment

File: app/src/main/java/io/plaidapp/ui/FeedAdapter.java [hunk 11]
```diff
@@ -115,6 +121,9 @@
     private ShotWeigher shotWeigher;
     private StoryWeigher storyWeigher;
     private PostWeigher postWeigher;
+    private DeviationWeigher deviationWeigher;
+
+//    private int deviantKeySize;
```
Comment (piotrek1543) [inline app/src/main/java/io/plaidapp/ui/FeedAdapter.java]: Please remove all commented out content

File: app/src/main/java/io/plaidapp/ui/DeviationActivity.java [hunk 12]
```diff
@@ -0,0 +1,1125 @@
...
+        if (deviantartPrefs.isLoggedIn()) {
+            if (TextUtils.isEmpty(enterComment.getText())) return;
+            enterComment.setEnabled(false);
+//            final Call<Comment> postCommentCall = deviantartPrefs.getApi().postComment(
+//                    shot.id, enterComment.getText().toString().trim());
+//            postCommentCall.enqueue(new Callback<Comment>() {
+//                @Override
+//                public void onResponse(Call<Comment> call, Response<Comment> response) {
+//                    loadComments();
+//                    enterComment.getText().clear();
```
Comment (piotrek1543) [inline app/src/main/java/io/plaidapp/ui/DeviationActivity.java]: Please remove all commented out content

File: app/src/main/java/io/plaidapp/data/DataManager.java [hunk 13]
```diff
@@ -428,6 +455,84 @@ public void onFailure(Call<List<Shot>> call, Throwable t) {
...
+            public void onFailure(Call<List<Deviation>> call, Throwable t) {
+                loadFailed(SourceManager.SOURCE_DEVIANTART_DAILY);
+            }
+        });
+        inflight.put(SourceManager.SOURCE_DEVIANTART_DAILY, userHandleCall);
+
+    }
+
+    private void loadPopularDeviations(final int page) {
+        // this API's paging is 0 based but this class (& sorting) is 1 based so adjust locally
```
Comment (piotrek1543) [inline app/src/main/java/io/plaidapp/data/DataManager.java]: Remove all commented out content

**Comment by piotrek1543** [comment]
@arniesaha check this link out: https://medium.com/google-developers/writing-more-code-by-writing-less-code-with-android-studio-live-templates-244f648d17c7

In a few comments I said about it. It really speed up your coding ;-)

**Comment by nickbutcher** [comment]
Thank you for the PR but this is not something we'd like to merge in the project. In the future please feel free to open an issue prior to developing a feature to discuss.

**Pull request closed without merging**